[front] - feat(skill): add icon selection

JulesBelveze opened the pull request:
## Description
Adds icon selection capability to the skill builder, allowing users to pick custom icons for their skills using the Sparkle `IconPicker` component. The icon field is stored in both `skill_configuration` and `skill_versions` tables.



<img width="1005" height="343" alt="Screenshot 2025-12-12 at 15 53 27" src="https://github.com/user-attachments/assets/462a6938-4d58-49f6-a058-51c0670492a9" />

**References**
- https://github.com/dust-tt/tasks/issues/5568

## Risk
Database schema migration adds nullable `icon` columns to two tables.

## Tests
Tested locally with skill creation and editing flows.

## Deploy Plan
1. Merge and deploy front
2. Run migration to add `icon` column to `skill_configuration` and `skill_versions` tables

JulesBelveze committed:
[skill_builder] - feature: add icon selection support for skill builder

 - Introduce an 'icon' field in the skill builder form schema to store the selected icon
 - Implement a new `SkillBuilderIconSection` component for the icon picker in the skill builder UI
 - Update the `SkillBuilderSettingsSection` to include the icon picker next to the skill name input
 - Modify the skill configuration transformation to include the new 'icon' field when converting to form data
 - Extend the skill model to persist the icon information in the database
 - Adjust the skill resource and API handlers to accept and process the 'icon' data
 - Ensure skill configurations are correctly updated with the new icon data when edited through the API
 - Provide default null value for the 'icon' field to support existing skills without icons

JulesBelveze committed:
[db] - feature: add icon column to skill_configuration table

 - A new 'icon' text column is added to the skill_configuration table to store iconography data associated with skill configurations

JulesBelveze committed:
[front/lib/models] - feature: add version tracking to SkillVersionModel

 - SkillVersionModel now includes a `version` property to keep track of version changes

[front/lib/resources/skill] - fix: ensure icon defaults to null if unspecified

 - SkillResource now explicitly sets `icon` to null when it's missing to maintain consistent data presentation

[front/migrations] - feature: extend skill tables with icon column

 - Migration script created to add an `icon` TEXT column to both `skill_configuration` and `skill_versions` tables

aubin-tchoi commented on front/lib/models/skill.ts:
```diff
@@ -91,6 +92,9 @@ SkillVersionModel.init(
     },
+    version: {
+      type: DataTypes.INTEGER,
+    },
```
should be icon string allowNull true

aubin-tchoi commented on front/lib/resources/skill/skill_resource.ts:
```diff
@@ -455,6 +349,160 @@ export class SkillResource extends BaseResource<SkillConfigurationModel> {
+    return SkillResource.fetchByIds(auth, allSkillIds);
+  }
+
+  private static async baseFetch(
```
was this method changed?

aubin-tchoi reviewed (approved):
LGTM

JulesBelveze committed:
[skill_builder] - refactor: standardize default icon handling

 - Extract the default icon to a constant for clarity and reusability
 - Ensure non-nullability of `version` field in SkillVersionModel

[models/skill] - feat: add `icon` field to Skill model

 - Introduce an optional `icon` field to store icon data for skills

[skill_resource] - refactor: reorder skill properties for consistency

 - Adjust the order of skill properties to maintain a consistent structure in SkillResource outputs

JulesBelveze committed:
[front/migrations] - fix: correct table name in migration script

 - Changed the table name from "skill_configuration" to "skill_configurations" to match the correct table schema
 - Added an "icon" TEXT column to the "skill_configurations" table to store icon information for skill configurations

JulesBelveze committed:
[front/skill] - fix: correct conditional check in SkillResource.update

 - Ensure that updates are only possible for non-global skill configurations by fixing incorrect conditional logic.

JulesBelveze committed:
[front/skill_builder] - refactor: sync editors only for existing skills

 - Exclude the editor synchronization logic during the creation of new skills
 - Editor sync now only occurs when updating existing skills, as new skills automatically add the creator as an editor

JulesBelveze committed:
[api/w/skills] - fix: add missing icon field to skill creation tests

 - Ensure that skill creation test payloads include the 'icon' field set to null to reflect recent schema changes
 - Update tests to align with the expected request structure for skill creation in the API endpoint

JulesBelveze committed:
[front] - test: add icon field with null value to skill update tests

 - Ensure that the icon field is included and set to null in skill update API tests
 - Reflect changes in the skill data structure in test cases to maintain test accuracy
